src/lib/concurrency/transaction_context.cpp

```diff
@@ -39,6 +39,12 @@ TransactionContext::~TransactionContext() {
                 return !has_registered_operators || committed_or_rolled_back;
               }()),
               "Has registered operators but has neither been committed nor rolled back.");
+
+  /**
+     * Tell the TransactionManager, which keeps track of active snapshot-commit-ids,
+     * that this transaction has finished.
+     */
+  TransactionManager::get().deregister_transaction(_snapshot_commit_id);
```

Comment:
This looks familiar. If this PR doesn't make sense before the other two are merged, let's wait for that. Otherwise, reviewers will have to look at the same things twice and consider changes that might be made in the other two PRs.

Reply:
Makes sense. I added a note at the top to clarify this circumstance.